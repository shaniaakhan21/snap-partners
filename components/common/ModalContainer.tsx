export const ModalContainer = ({ children }) => {
  return (
    <div
      className='w-[calc(100vw-20px)] max-w-[500px] max-h-[calc(100vh-60px)] rounded-sm shadow-xl bg-white p-4'
    >
      {children}
    </div>
  )
}
