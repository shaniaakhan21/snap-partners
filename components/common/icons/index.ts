export { AdjuntIcon } from './Adjunt'
export { ArrowDownIcon } from './ArrowDown'
export { ArrowHeadRightIcon } from './ArrowHeadRight'
export { ArrowRightIcon } from './ArrowRight'
export { CalendarIcon } from './Calendar'
export { CheckIcon } from './Check'
export { CompensationPlanIcon } from './CompensationPlan'
export { CopyIcon } from './Copy'
export { CustomerIcon } from './Customer'
export { DashboardIcon } from './Dashboard'
export { DownloadIcon } from './Download'
export { DriverIcon } from './Driver'
export { EmailIcon } from './Email'
export { EyeHiddenIcon } from './EyeHidden'
export { EyeVisibleIcon } from './EyeVisible'
export { FacebookColorIcon } from './FacebookColor'
export { FacebookIcon } from './Facebook'
export { GenealogyIcon } from './Genealogy'
export { GoogleColorIcon } from './GoogleColor'
export { GoogleIcon } from './Google'
export { IBOIcon } from './IBO'
export { LinkedinIcon } from './Linkedin'
export { MarketingIcon } from './Marketing'
export { MerchantIcon } from './Merchant'
export { MoarOptionsVerticalIcon } from './MoarOptionsVertical'
export { NotificationIcon } from './Notification'
export { PhoneIcon } from './Phone'
export { ReferralLinksIcon } from './ReferralLinks'
export { RestaurantsIcon } from './Restaurants'
export { SearchIcon } from './Search'
export { ShareIcon } from './Share'
export { ShareRRSSIcon } from './ShareRRSS'
export { StarIcon } from './Star'
export { TrendingUpIcon } from './TrendingUp'
export { TwitterColorIcon } from './TwitterColor'
export { TwitterIcon } from './Twitter'
