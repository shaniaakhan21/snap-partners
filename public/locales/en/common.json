{"agreeTo":"Agree to","termsAndConditions":"Terms and Conditions"}