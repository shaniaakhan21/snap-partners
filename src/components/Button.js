import React from 'react'

export const Button = ({
  children,
  onClick = () => console.log('onClick Default'),
  disabled = false,
  type = 'button',
  classes = '',
  style
}) => (
  <button
    type={type}
    disabled={disabled}
    className={`px-4 py-2 disabled:opacity-50 disabled:cursor-not-allowed bg-black-primary text-white bg-primary-500 rounded-full font-semibold focus:outline-none hover:opacity-90 ${classes}`}
    onClick={onClick}
    style={style}
  >
    {children || 'BUTTON DEFAULT'}
  </button>
)
