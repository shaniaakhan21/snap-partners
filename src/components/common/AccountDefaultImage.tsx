import { IAuth } from 'lib/stores/Auth'
import { TRANK } from 'lib/types/user/ranks';

interface AccountDefaultImageProps {
  rank: TRANK;
  size?: number | string
}
export default function AccountDefaultImage (props: AccountDefaultImageProps) {
  const { rank, size = '140px' } = props
  const imgPath = `/images/profile/${rank}.png`

  return <img src={imgPath} style={{ width: size, height: size, borderRadius: size }}/>
}
