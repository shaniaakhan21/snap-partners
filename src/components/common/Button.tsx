import { ReactNode, CSSProperties } from 'react'

type ButtonProps = {
  children?: ReactNode
  onClick?: (params: any) => void
  disabled?: boolean
  type?: 'submit' | 'reset' | 'button'
  classes?: String
  style?: CSSProperties
}

export const Button = ({
  children,
  onClick = () => console.log('onClick Default'),
  disabled = false,
  type = 'button',
  classes = '',
  style
}: ButtonProps) => (
  <button
    type={type}
    disabled={disabled}
    className={`px-4 py-2 disabled:opacity-50 disabled:cursor-not-allowed -primary text-white bg-primary-500 rounded-full font-semibold focus:outline-none focus:ring focus:ring-primary-300 focus:opacity-90 hover:opacity-90 ${classes}`}
    onClick={onClick}
    style={style}
  >
    {children || 'BUTTON DEFAULT'}
  </button>
)
