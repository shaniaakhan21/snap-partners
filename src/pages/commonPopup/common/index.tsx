import { Button, Modal } from '@mui/material'
import { Close as CrossIcon } from '@mui/icons-material'
import Link from 'next/link';

interface PopupProps {
  open: boolean;
  onClose: () => void;
  image: string;
  title: string;
  description: string;
  buttonText: string;
  svgId: string;
}

const CommonPopup = ({ open, onClose, image, title, description, buttonText, svgId }: PopupProps) => {
  return (
    <Modal open={open} onClose={onClose} className='overflow-y-scroll'>
      <div className='w-full flex justify-center outline-none'>
        <div className='bg-white rounded-xl p-4 px-10 pb-10 w-6/12 my-20 ml-10'>
          <div className='flex flex-row justify-end'>
            <div>
              <CrossIcon onClick={onClose} className='text-5xl cursor-pointer' />
            </div>
          </div>
          <div className='flex flex-col justify-center text-center'>
            <div className='flex w-full justify-center m-1'>
              <img src={image} alt="Popup Image" id={svgId} />
            </div>
            <h3 className='text-2xl font-semibold mt-2'>{title}</h3>
            <h5 className='font-semibold text-[#878787]'>{description}</h5>
          </div>
          <br />
          <div className='w-full flex justify-center'>
            <Button
              type="submit"
              variant="contained"
              className='send-button text-xl rounded-xl text-center px-8 capitalize py-4 text-base bg-primary-500'
              onClick={() => {
                window.location.reload()
              }}
            ><Link href='/' >
                {buttonText}
              </Link>
            </Button>
          </div>
        </div>
      </div>
    </Modal>
  )
}

export default CommonPopup
