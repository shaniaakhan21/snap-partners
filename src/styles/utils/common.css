.align-center{
    text-align: center;
}

.align-end{
    text-align: end;
}

.red-text{
    color: #E35C49;
}

.disc-here::before{
        content: 'â€¢';
        font-size: 16px; 
        margin-right: 10px; 
        display: inline-block;
        vertical-align: middle
}

.custom-green{
    background-color: rgba(84, 165, 44, 1);
}

.custom-yellow{
    background-color: #FFAA00;
}

.custom-gray{
    background-color: #404040;
}

.myDataGrid {
    @media screen and (max-width: 480px) {
          width: 120%;
        }
}

.myDataGridSum {
    @media screen and (max-width: 480px) {
          width: 300%;
        }
}

.myDataGridIT {
    @media screen and (max-width: 480px) {
          width: 600%;
        }
}

.myDataGridTB {
    @media screen and (max-width: 480px) {
          width: 900%;
        }
}
.myDataGrid .MuiDataGrid-columnHeader {
    border-right: 'none!important';
    font-weight: 'bold!important';
    font-size: '1.5em!important';
}
.myDataGrid .MuiDataGrid-cell {
    border-right: 'none!important';
}

.left-83{
    left: 83%;
    @media screen and (max-width: 399px) {
        left: 0;
    }
}