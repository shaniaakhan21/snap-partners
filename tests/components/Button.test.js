/* eslint-disable no-undef */
import { render } from '@testing-library/react'
import { Button } from 'components/common/Button'

describe('Test Button Component', () => {
  it('Should render Button unchanged', () => {
    const buttonTest = <Button>My Button Test</Button>

    const { container } = render(buttonTest)
    expect(container).toMatchSnapshot()
  })
})
