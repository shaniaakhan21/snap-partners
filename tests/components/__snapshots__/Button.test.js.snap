// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Component Should render Button unchanged 1`] = `
<div>
  <button
    class="px-4 py-2 disabled:opacity-50 disabled:cursor-not-allowed bg-black-primary text-white bg-primary-500 rounded-full font-semibold focus:outline-none hover:opacity-90 "
    type="button"
  >
    My Button Test
  </button>
</div>
`;
